src/ql/pass/map/qubits/place_mip/detail/impl.h

```diff
@@ -0,0 +1,163 @@
+#pragma once
+
+#include "ql/utils/num.h"
+#include "ql/utils/str.h"
+#include "ql/utils/ptr.h"
+#include "ql/utils/vec.h"
+#include "ql/utils/pairhash.h"
+#include "ql/ir/ir.h"
+#include "ql/ir/compat/compat.h"
+#include "ql/com/map/qubit_mapping.h"
+
+class HighsModel;
+
+namespace ql {
+namespace pass {
+namespace map {
+namespace qubits {
+namespace place_mip {
+namespace detail {
+
+/**
+ * Options structure for configuring the initial placement algorithm.
+ */
+struct Options {
+    /**
+     * Filename where to write the MPS model.
+     */
+    utils::Str model_filename = "";
+
+    /**
+     * Whether to write the MPS model to a file.
+     */
+    utils::Bool write_model_to_file = true;
+
+    /**
+     * Timeout for the MIP algorithm in seconds, or 0 to disable timeout.
+     */
+    utils::Real timeout = 0.0;
+
+    /**
+     * The placement algorithm will only consider the connectivity required to
+     * perform the first horizon two-qubit gate types. 0 means that all
+     * gate types should be considered.
+     */
+    utils::UInt horizon = 0;
+};
+
```

Comment:
Furthermore, the old system had an option to choose what to do on timeout: ignore the mapping and continue, or let the compiler stop. With this new system, it could also be an option to continue with the mapping that is there at the time out.

Reply:
Added fail_on_timeout